all_test.go

```diff
@@ -2930,6 +2930,88 @@ func TestMin(t *testing.T) {
 	}
```

Comment:
I think one or more tests in this file are leaking/sharing state which prevents the entire file from passing when all the tests are run at once. However, all the max/min tests pass as far as I can tell:
```
$ go test -run "^TestMaxInt|TestMinInt|TestMaxUint|TestMax|TestMin|TestMaxVal|TestMinVal|TestClamTestUMax|TestUMin|TestUMaxVal|TestUMinVal|TestUClamTestMaxByte|TestMinByte|TestMaxByteVal|TestMinByteVal|TestMaxUint16|TestMinUint16|TestMaxUint16Val|TestMinUint16Val|TestMaxUint32|TestMinUint32|TestMaxUint32Val|TestMinUint32Val|TestMaxUint64|TestMinUint64|TestMaxUint64Val|TestMinUint64Val|TestMaxInt8|TestMinInt8|TestMaxInt8Val|TestMinInt8Val|TestMaxInt16|TestMinInt16|TestMaxInt16Val|TestMinInt16Val|TestMaxInt32|TestMinInt32|TestMaxInt32Val|TestMinInt32Val|TestMaxInt64|TestMinInt64|TestMaxInt64Val|TestMinInt64Val$"
PASS
ok  	_/Users/bodeckerdellamaria/Documents/Comp_Sci/mathutil	0.009s
```

Reply:
> I think one or more tests in this file are leaking/sharing state which prevents the entire file from passing when all the tests are run at once. 

I see no failing tests, please explain.

```
jnml@4670:~/src/github.com/cznic/mathutil$ git branch 
* master
jnml@4670:~/src/github.com/cznic/mathutil$ git log -1
commit 53c70789c7ff53526fd6c28c5a6881b138413e0c
Merge: 91af0ce 2eb44b6
Author: Muhammad Surya <surya.asriadie@gmail.com>
Date:   Mon Sep 25 19:14:14 2017 +0700

    Merge pull request #8 from Fs02/master
    
    Support for Clamp
jnml@4670:~/src/github.com/cznic/mathutil$ go test
PASS
ok  	github.com/cznic/mathutil	23.017s
jnml@4670:~/src/github.com/cznic/mathutil$ git checkout -b bojdell-bojdell/add-variadic-max-min master
Switched to a new branch 'bojdell-bojdell/add-variadic-max-min'
jnml@4670:~/src/github.com/cznic/mathutil$ git pull https://github.com/bojdell/mathutil.git bojdell/add-variadic-max-min
remote: Counting objects: 9, done.
remote: Compressing objects: 100% (3/3), done.
remote: Total 9 (delta 6), reused 9 (delta 6), pack-reused 0
Unpacking objects: 100% (9/9), done.
From https://github.com/bojdell/mathutil
 * branch            bojdell/add-variadic-max-min -> FETCH_HEAD
Updating 53c7078..2d65390
Fast-forward
 CONTRIBUTORS |   1 +
 all_test.go  | 624 +++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++
 mathutil.go  | 227 +++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++
 3 files changed, 852 insertions(+)
jnml@4670:~/src/github.com/cznic/mathutil$ go test |& tee log
PASS
ok  	github.com/cznic/mathutil	22.966s
jnml@4670:~/src/github.com/cznic/mathutil$ 
```

Reply:
Ah, must be the timeout the Go VSCode plugin uses when running tests.
```
Running tool: /usr/local/go/bin/go test -timeout 30s -run ^Test0|Test1|Test2|TestBig0|TestBig1|TestBig2|TestPermutations|TestIsPrime|TestNextPrime|TestNextPrime2|TestNextPrimeUint64|TestISqrt|TestSqrtUint64|TestSqrtBig|TestFactorInt|TestFactorIntB|TestIsPrimeUint16|TestNextPrimeUint16|TestPopCount|TestGCD|TestLog2|TestBitLen|TestUintptrBits|TestModPowByte|TestModPowUint16|TestModPowUint32|TestModPowUint64|TestModPowBigInt|TestAdd128|TestMul128|TestIsPrimeUint64|TestProbablyPrimeUint32|TestProbablyPrimeUint64_32|TestProbablyPrimeBigInt_32|TestProbablyPrimeBigInt|TestQCmpUint32|TestQScaleUint32|TestPrimorialProductsUint32|TestPowerizeUint32BigInt|TestPowerizeBigInt|TestEnvelope|TestMaxInt|TestMinInt|TestMaxUint|TestMax|TestMin|TestMaxVal|TestMinVal|TestClamp|TestUMax|TestUMin|TestUMaxVal|TestUMinVal|TestUClamp|TestMaxByte|TestMinByte|TestMaxByteVal|TestMinByteVal|TestClampByte|TestMaxUint16|TestMinUint16|TestMaxUint16Val|TestMinUint16Val|TestClampUint16|TestMaxUint32|TestMinUint32|TestMaxUint32Val|TestMinUint32Val|TestClampUint32|TestMaxUint64|TestMinUint64|TestMaxUint64Val|TestMinUint64Val|TestClampUint64|TestMaxInt8|TestMinInt8|TestMaxInt8Val|TestMinInt8Val|TestClampInt8|TestMaxInt16|TestMinInt16|TestMaxInt16Val|TestMinInt16Val|TestClampInt16|TestMaxInt32|TestMinInt32|TestMaxInt32Val|TestMinInt32Val|TestClampInt32|TestMaxInt64|TestMinInt64|TestMaxInt64Val|TestMinInt64Val|TestClampInt64|TestPopCountBigInt|TestToBase|TestBug|TestQuadPolyDiscriminant|TestQuadPolyFactors|TestFCPRNG$

panic: test timed out after 30s

goroutine 20 [running]:
testing.startAlarm.func1()
	/usr/local/go/src/testing/testing.go:1145 +0xf9
created by time.goFunc
	/usr/local/go/src/time/sleep.go:170 +0x44

goroutine 1 [chan receive]:
testing.(*T).Run(0xc4200fc000, 0x1183437, 0x17, 0x1189908, 0x1068a01)
	/usr/local/go/src/testing/testing.go:790 +0x2fc
testing.runTests.func1(0xc4200fc000)
	/usr/local/go/src/testing/testing.go:1004 +0x64
testing.tRunner(0xc4200fc000, 0xc42009dde0)
	/usr/local/go/src/testing/testing.go:746 +0xd0
testing.runTests(0xc42000a140, 0x1266600, 0x64, 0x64, 0xf3635fc3f32b0339)
	/usr/local/go/src/testing/testing.go:1002 +0x2d8
testing.(*M).Run(0xc420311f18, 0xc42009df70)
	/usr/local/go/src/testing/testing.go:921 +0x111
main.main()
	_/Users/bodeckerdellamaria/Documents/Comp_Sci/mathutil/_test/_testmain.go:472 +0xdb

goroutine 66 [runnable]:
math/big.(*Int).QuoRem(0xc42004cc88, 0xc4202f6240, 0xc4204bad40, 0xc4202f6240, 0xc42004cc88, 0xc4202f6240)
	/usr/local/go/src/math/big/int.go:226 +0x21c
math/big.(*Int).Mod(0xc4202f6240, 0xc4202f6240, 0xc4204bad40, 0xc4202f6240)
	/usr/local/go/src/math/big/int.go:258 +0x173
_/Users/bodeckerdellamaria/Documents/Comp_Sci/mathutil.modPowBigInt(0xc4202f6240, 0xc4202f6260, 0xc4204bad40, 0xc4202f58c0)
	/Users/bodeckerdellamaria/Documents/Comp Sci/mathutil/mathutil.go:326 +0xbf
_/Users/bodeckerdellamaria/Documents/Comp_Sci/mathutil.ModPowBigInt(0xc42004cef0, 0xc42004ce60, 0xc4204bad40, 0xc42004ce60)
	/Users/bodeckerdellamaria/Documents/Comp Sci/mathutil/mathutil.go:345 +0x252
_/Users/bodeckerdellamaria/Documents/Comp_Sci/mathutil.ProbablyPrimeBigInt(0xc4204bad40, 0xc42004cef0, 0xc42004cef0)
	/Users/bodeckerdellamaria/Documents/Comp Sci/mathutil/mathutil.go:606 +0x265
_/Users/bodeckerdellamaria/Documents/Comp_Sci/mathutil.TestProbablyPrimeBigInt.func1(0x8b429, 0x14f5d5, 0xc420018110, 0x2, 0x2)
	/Users/bodeckerdellamaria/Documents/Comp Sci/mathutil/all_test.go:2185 +0x110
_/Users/bodeckerdellamaria/Documents/Comp_Sci/mathutil.TestProbablyPrimeBigInt(0xc4200fc1e0)
	/Users/bodeckerdellamaria/Documents/Comp Sci/mathutil/all_test.go:2210 +0x1b5
testing.tRunner(0xc4200fc1e0, 0x1189908)
	/usr/local/go/src/testing/testing.go:746 +0xd0
created by testing.(*T).Run
	/usr/local/go/src/testing/testing.go:789 +0x2de
exit status 2
FAIL	_/Users/bodeckerdellamaria/Documents/Comp_Sci/mathutil	30.018s
Error: Tests failed.
```

I don't think it's an issue. When I run from the terminal, all tests pass fine:
```
bodeckerdellamaria@Bodeckers-MBP mathutil (bojdell/add-variadic-max-min) $ go test
PASS
ok  	_/Users/bodeckerdellamaria/Documents/Comp_Sci/mathutil	34.392s
bodeckerdellamaria@Bodeckers-MBP mathutil (bojdell/add-variadic-max-min) $ git log
commit 2d65390f2b0b6583cbd637596199593b5eed73d5
Author: Bodecker DellaMaria <bojdell@gmail.com>
Date:   Sat Oct 14 15:54:08 2017 -0700

    CONTRIBUTORS: update

    Updates the contributors list.
```
Sorry for the confusion!